test/helper.sh

```diff
@@ -0,0 +1,216 @@
+        if [[ "${certStatus}" != "Active" ]]; then
+            sleep 60
+            continue
+        fi
+
+        # Check status of each domain.
+        domainStatus=( $(kubectl get ManagedCertificate ${mc} -n ${ns} -o jsonpath="{.status.domainStatus}" | jq  ".[] | .status") )
+        count=${#domainStatus[@]}
+        match=0
+        for status in "${domainStatus[@]}"; do
+            if [[ "${status}" == '"Active"' ]]; then
+                ((match++))
+            fi
+        done
+
+        if [[ "${match}" -eq "${count}" ]]; then
+            return
+        fi
+        sleep 60
+    done
+    exit 1
+}
+
+###############################################################################
+# Basic setup that works for most recipe tests.
+# Globals:
+#   ZONE
+#   SUBNET_REGION
+# Arguments:
+#     Name of the test. Used to generate suffix for resources.
+###############################################################################
+setup_gke_basic() {
+    local test_name
+    test_name="$1"
+    resource_suffix=$(get_hash "${test_name}")
+    resource_name="gke-net-recipes-${resource_suffix}"
+
+    gcloud compute networks create "${resource_name}" --subnet-mode="custom"
+    gcloud compute networks subnets create "${resource_name}" --network="${resource_name}" --region="${SUBNET_REGION}" --range="10.1.2.0/24"
+    gcloud compute instances create "${resource_name}" --zone="${ZONE}" --network="${resource_name}" --subnet="${resource_name}" --image-family="debian-11" --image-project="debian-cloud" --tags="allow-ssh"
+    gcloud container clusters create "${resource_name}" --zone="${ZONE}" --network="${resource_name}" --subnetwork="${resource_name}"
+    gcloud container clusters get-credentials "${resource_name}" --zone="${ZONE}"
+}
+
+###############################################################################
+# Cleanup the basic setup for recipe test.
+# Globals:
+#   ZONE
+#   SUBNET_REGION
+# Arguments:
+#     Name of the test. Used to generate suffix for resources.
+###############################################################################
+cleanup_gke_basic() {
+    local test_name
+    test_name="$1"
+    resource_suffix=$(get_hash "${test_name}")
+    resource_name="gke-net-recipes-${resource_suffix}"
+
+    gcloud container clusters delete "${resource_name}" --zone="${ZONE}" --quiet || true
+    gcloud compute instances delete "${resource_name}" --zone="${ZONE}" --quiet || true
+    gcloud compute networks subnets delete "${resource_name}" --region="${SUBNET_REGION}" --quiet || true
+
+    # Cleanup the firewalls associated with the network before deleting it.
+    firewalls=( $(gcloud compute firewall-rules list --filter="network=\"${resource_name}\"" --format="value(NAME)") )
+    for fw in "${firewalls[@]}"; do
+        echo "${fw}" | awk '{printf "gcloud compute firewall-rules delete %s --quiet\n", $1}' | bash || true
```

Comment:
Can you explain what the extra level of running `awk` is doing here?

Why can't we do:

```
for fw in "${firewalls[0]}"; do
  gcloud compute firewall-rules delete "${fw}" --quiet || true
done
```